distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/strategy/EvenlySpreadAssignmentStrategy.java

```diff
@@ -0,0 +1,62 @@
+package ru.fix.distributed.job.manager.strategy;
+
+import ru.fix.distributed.job.manager.model.AssignmentState;
+import ru.fix.distributed.job.manager.model.JobId;
+import ru.fix.distributed.job.manager.model.WorkItem;
+import ru.fix.distributed.job.manager.model.WorkerId;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class EvenlySpreadAssignmentStrategy extends AbstractAssignmentStrategy {
+
+    @Override
+    public AssignmentState reassignAndBalance(
+            Map<JobId, Set<WorkerId>> availability,
+            AssignmentState prevAssignment,
+            AssignmentState currentAssignment,
+            Set<WorkItem> itemsToAssign
+    ) {
+        for (Map.Entry<JobId, Set<WorkerId>> jobEntry : availability.entrySet()) {
+            Set<WorkItem> itemsToAssignForJob = getWorkItemsByJob(jobEntry.getKey(), itemsToAssign);
+            jobEntry.getValue().forEach(e -> currentAssignment.putIfAbsent(e, new HashSet<>()));
+
+            int workersCount = jobEntry.getValue().size();
+            int workItemsCount = itemsToAssignForJob.size();
+            int limitWorkItemsOnWorker = limitWorkItemsOnWorker(workItemsCount, workersCount);
+            int itemsAssignedFromPreviousCounter = 0;
+
+            for (WorkItem item : itemsToAssignForJob) {
+                WorkerId workerFromPrevious = prevAssignment.getWorkerOfWorkItem(item);
+
+                if (prevAssignment.containsWorkItem(item)
+                        && currentAssignment.containsKey(workerFromPrevious)
+                        && currentAssignment.isBalanced()
+                        && currentAssignment.isBalancedByJobId(item.getJobId())
+                        && itemsAssignedFromPreviousCounter < limitWorkItemsOnWorker
+                ) {
+                    currentAssignment.addWorkItem(workerFromPrevious, item);
+                    itemsAssignedFromPreviousCounter++;
```

Comment:
too complex, probably better to discuss verbally and write comment  with clarification here.  
Also there is a suggestion for simple evenly spread algorithm. Let's discuss.

Reply:
I simplified algorithm:
Now it works like this:
1) All work items are transferred from the previous state, if they were there. If they were not, then add to the less loaded.
2) We balance the local work pools by transferring the work item from the most loaded to the less loaded
3) Balancing the global work pool